browser/brave_profile_prefs.cc

```diff
@@ -39,7 +39,9 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   // Default Brave shields
   registry->RegisterBooleanPref(kHTTPSEVerywhereControlType, true);
   registry->RegisterBooleanPref(kNoScriptControlType, false);
+}
 
+void OverrideDefaultProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
```

Comment:
it seems to me that if we have to avoid setting our default values to get the test to pass then the test doesn't work in Brave

Reply:
We can't set default values for unregistered preferences. The correct way to fix this is to ensure that all preferences we want to override are registered beforehand, but this causes much bigger diff. I think we can return to this issue when the process of enabling chromium tests for Brave starts.

Reply:
which prefs aren't registered? Can we split them out?

Reply:
some of them are probably dependent on other services that don't get registered in testing so we can also add checks by using `ServiceIsNULLWhileTesting` before overriding

Reply:
first two or three, I didn't check others. Basically, if we want to override defaults for the testing profile (probably in another PR) we can add this call to the end of `TestingProfile::Init()`, where services prefs should be already registered. This may solve the issue. And maybe add some checks then.

Reply:
I'm fine with moving to a follow-up, but please create an issue to track

Reply:
I just worry that we're not getting a realistic test here because some of these prefs affect startup/newtab behavior